packetbeat/magefile.go

```diff
@@ -131,33 +139,85 @@ func TestPackages() error {
 	return mage.TestPackages()
 }
 
-// Update updates the generated files (aka make update).
-func Update() error {
-	return sh.Run("make", "update")
+// Update updates the generated files.
+func Update() {
+	mg.SerialDeps(Fields, Dashboards, Config, includeList, fieldDocs)
+}
+
+// Config generates the config files.
+func Config() error {
+	return mage.Config(mage.AllConfigTypes, packetbeat.ConfigFileParams(), ".")
 }
 
-// Fields generates a fields.yml and fields.go for the Beat.
+func includeList() error {
+	return mage.GenerateIncludeListGo([]string{"protos/*"}, nil)
+}
+
+// Fields generates fields.yml and fields.go files for the Beat.
 func Fields() {
-	mg.SerialDeps(fieldsYML, mage.GenerateAllInOneFieldsGo)
+	mg.Deps(libbeatAndPacketbeatCommonFieldsGo, protosFieldsGo)
+	mg.Deps(fieldsYML)
 }
 
-// fieldsYML generates the fields.yml file.
+// libbeatAndPacketbeatCommonFieldsGo generates a fields.go containing both
+// libbeat and packetbeat's common fields.
+func libbeatAndPacketbeatCommonFieldsGo() error {
+	if err := mage.GenerateFieldsYAML(); err != nil {
+		return err
+	}
+	return mage.GenerateAllInOneFieldsGo()
+}
+
+// protosFieldsGo generates a fields.go for each protocol.
+func protosFieldsGo() error {
+	return mage.GenerateModuleFieldsGo("protos")
+}
+
+// fieldsYML generates the fields.yml file containing all fields.
 func fieldsYML() error {
 	return mage.GenerateFieldsYAML("protos")
 }
 
-// GoTestUnit executes the Go unit tests.
+func fieldDocs() error {
+	return mage.Docs.FieldDocs("fields.yml")
+}
+
+// Dashboards collects all the dashboards and generates index patterns.
+func Dashboards() error {
+	return mage.KibanaDashboards("protos")
+}
+
+// Fmt formats source code and adds file headers.
+func Fmt() {
+	mg.Deps(mage.Format)
```

Comment:
Will it be possible to inherit these at one stage? I remember there was an issue for it in mage.

Reply:
I was looking into this tonight. There has been progress. See https://magefile.org/importing/